Pull eth_bots out of examples (#776)
delvtech/elf-simulations: eth_bots/README.md

# Running Bots

We have set up bots to run on a dedicated AWS EC2 instance.
These instructions can likely be followed for other unix/linux setups, however.

## Step 1: Install elfpy

1. make a fork of the [delvtech/elf-simulations repo](https://github.com/delvtech/elf-simulations) ([GitHub fork instructions](https://docs.github.com/en/get-started/quickstart/fork-a-repo?tool=webui&platform=mac)).

2. Install elfpy by following the installation instructions found [on github](https://github.com/delvtech/elf-simulations/blob/main/INSTALL.md).

    >**💡NOTE:**
    >
    >**pyenv install tips:**
    >

>     >If your using an AWS server running bash on Linux, then to install Pyenv you’ll want to follow the [automatic installer](https://github.com/pyenv/pyenv#automatic-installer) instructions, and then [add pyenv to your shell environment](https://github.com/pyenv/pyenv#set-up-your-shell-environment-for-pyenv).

Reviewer: ```suggestion
    >If you're using an AWS server running bash on Linux, then to install Pyenv you’ll want to follow the [automatic installer](https://github.com/pyenv/pyenv#automatic-installer) instructions, and then [add pyenv to your shell environment](https://github.com/pyenv/pyenv#set-up-your-shell-environment-for-pyenv).
```

Author: good catch!

    >
    >**git clone the correct repo:**
    >

>     >You’ll want to clone your fork in step 2 of the install instructions.

Reviewer: ```suggestion
    >You’ll want to clone your fork
```

# Running Bots

We have set up bots to run on a dedicated AWS EC2 instance.
These instructions can likely be followed for other unix/linux setups, however.

## Step 1: Install elfpy

1. make a fork of the elf `delvtech/elf-simulations` repo ([instructions](https://docs.github.com/en/get-started/quickstart/fork-a-repo?tool=webui&platform=mac)).

    >**💡NOTE:** 

>     >You’ll probably want to configure the remote (`delvtech`) repository with your fork. Once you’ve made the fork and cloned it, navigate inside (`cd elf-simulations`), and run

Reviewer: ```suggestion
    >If you want to save your changes, [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) the remote (`delvtech`) and clone your fork instead. Then, navigate inside (`cd elf-simulations`), and run
```

    >
    >```bash
    >git remote add upstream git@github.com:delvtech/elf-simulations.git
    >```
    >
    >to sync up an `upstream` repository.
    >Then whenever you want to sync your fork with the latest code you can run:
    >
    >```bash
    >git fetch upstream
    >git merge upstream/main
    >```
    >

    
2. Install elfpy by following the installation instructions found [on github](https://github.com/delvtech/elf-simulations/blob/main/INSTALL.md).
    
    >**💡NOTE:**
    >

>     >**pyenv:** If your the AWS server is running bash on linux, then to install Pyenv you’ll want to follow the “[automatic installer](https://github.com/pyenv/pyenv#automatic-installer)” instructions, which tells you to run `curl https://pyenv.run | bash` in your AWS terminal.

Reviewer: ```suggestion
    >Elf-simulations requires python 3.10. We recommend Pyenv to easily manage multiple python versions, which you can install with the “[automatic installer](https://github.com/pyenv/pyenv#automatic-installer)” instructions, which tells you to run `curl https://pyenv.run | bash` in your AWS terminal.
```

>     >Per the output of that command when the install is finished, you’ll want to add these lines to the bottom of your `~/.bashrc`:

Reviewer: ```suggestion
    >[Add Pyenv to your shell](https://github.com/pyenv/pyenv#set-up-your-shell-environment-for-pyenv). If using bash, that means adding these lines to the bottom of your `~/.bashrc`:
```

    
    </aside>
    
3. navigate to the `elf-simulations` folder: `cd elf-simulations/`
4. run the elfpy tests to verify that everything installed correctly by executing `python -m pytest`. Make sure you have enabled the correct Python environment!

# Step 2: Fund your bots & start trading!

1. get your private key for the chain (e.g. from Anvil)
2. modify `eth_bots/eth_bots_config.py` as you see fit for your experiment.
    
    <aside>
    💡 Make sure you change the URLs (e.g. to AWS or `localhost`):
    
    ```python
    username_register_url="http://<AWS_IP>:<UNAME_PORT>"
    artifacts_url="http://<AWS_IP>:<ARTIFACTS_PORT>"
    rpc_url="http://<AWS_IP>:<RPC_PORT>"
    ```
    
    </aside>
    
3. run the `eth_bots/populate_env.py` script with your private key as an argument, and pipe the output to a `.env` file. For example:
    
    <aside>
    💡 **NOTE:** This command overwrites your `.env` file, so make sure you're ok with losing whatever contents are in it first!
    
    ```bash
    python eth_bots/populate_env.py 0xUSER_PRIVATE_KEY > .env
    ```
    
    This will generate new environment variables for the bots and write them to the `.env` file.

>     The new variables are private keys, Base budgets, and Eth budgets for all of the agents you specified in your config.

Reviewer: ```suggestion
    The new variables are private keys, randomly generated Base and Eth budgets (which you can manually overwrite) for all of the agents you specified in your config.
```
